### DISCUSSION TITLE
Customising and packaging mkdocs-material

### DISCUSSION THREAD

**Discussion by SoumayaMauthoorMOJ** [squidfunk/mkdocs-material#4226] [General]
I would like to customise mkdocs-material to fit with our organisation design standards and package it so that it can be applied to other repos as a theme. mkdocs has generic material about packaging themes : https://www.mkdocs.org/dev-guide/themes/#packaging-themes. Do you have material about packaging themes based on mkdocs-material?

**Comment by squidfunk** [comment]
It's probably a better idea to ask the maintainers of MkDocs, since your question generalizes to packaging customizations for an existing MkDocs theme. I think it should be possible to package an extended theme, but I've never done it.

**Reply by SoumayaMauthoorMOJ** [reply]
Thanks for letting me know will ask!

**Comment by sumau** [comment]
Just in case anyone else interested,  you can package customizations of existing MKdocs theme the same way that you can [package themes](https://www.mkdocs.org/dev-guide/themes/#packaging-themes). You just need to extend the existing theme in the mkdocs_theme.yml file:
```
extends: material
```
Note that the instructions for packaging themes is a bit out of date. My [project](https://github.com/ministryofjustice/mkdocs-tech-docs-template) uses pyproject.toml and setup.cfg following the approach in this [project](https://git.ligo.org/computing/igwn-material/-/tree/main/mkdocs), but you could use other [build backends](https://packaging.python.org/en/latest/tutorials/packaging-projects/#creating-pyproject-toml)
